modules/local/summarizer.nf

```diff
@@ -1,21 +1,28 @@
 process SUMMARIZER {
+    tag "$meta.id"
+    label 'process_single'
 
+    conda "conda-forge::python=3.10.4 pandas=1.5.2"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
         'https://depot.galaxyproject.org/singularity/pandas:1.5.2' :
         'biocontainers/pandas:1.5.2' }"
     input:
-    path(tosummarize)
+    tuple val(meta), path(tosummarize)
 
     output:
-    path("summary.tsv"), emit: summary
+    tuple val(meta), path("summary.tsv"), emit: summary
     path("versions.yml"), emit: versions
 
+    when:
+    task.ext.when == null || task.ext.when
+
     script:
     """
     #!/usr/bin/env python
     import glob
     import pandas as pd
     import subprocess
+    import numpy
 
     files_kraken2 = glob.glob('*.kraken2_summary.tsv')
     files_blastn = glob.glob('*.blastn_summary.tsv')
```

Comment:
```suggestion
    files_blastn = glob.glob('*.blastn_summary.tsv')
    
    kraken2_dfs = [pd.read_csv(file, sep="\\t") for file in files_kraken2]
    df_kraken2 = pd.concat(kraken2_dfs)
    
    blastn_dfs = [pd.read_csv(file, sep="\\t") for file in files_blastn]
    df_blastn = pd.concat(blastn_dfs)

    summary_df = pd.concat([df_kraken2, df_blastn.reindex(df_kraken2.index)],axis=1)
    summary_df.to_csv("summary.tsv", sep="\\t"
   ```
I'm not quite sure how the dfs look that you are merging but the operation concat(..,axis=1) is usually done using join() (also as you overwrite the blast index with the kraken2 index). Also pandas is using index_col=0 anyway so no need to use it here.

Reply:
this leads to an output like that:

```
	Unnamed: 0	kraken2	isolatedkraken2	Unnamed: 0	blastn_unique_ids	blastn_lines	filteredblastn_unique_ids	filteredblastn_lines
0	test_minigut_forward_hg38host	25993	25993	test_minigut_forward_hg38host	11	11	11	11
0	test_bacass_long	75	75	test_minigut_forward_reverse_hg38host	17	17	17	17
0	test_minigut_forward_reverse_hg38host	25992	25991	test_bacass_long	0	0	0	0

```
I want it to be like this:

```
	kraken2	isolatedkraken2	blastn_unique_ids	blastn_lines	filteredblastn_unique_ids	filteredblastn_lines
test_minigut_forward_hg38host	25993	25993	11	11	11	11
test_bacass_long	75	75	0	0	0	0
test_minigut_forward_reverse_hg38host	25992	25991	17	17	17	17
```

If there is a way to improve the script without letting go of the wished format, let me know :)
But I am aware that the format could be improved, not decided yet how to improve to gain better human AND machine readability (maybe yaml per line or so).

Reply:
```suggestion
     files_blastn = glob.glob('*.blastn_summary.tsv')
    
    kraken2_dfs = [pd.read_csv(file, sep="\\t", index_col=0) for file in files_kraken2]
    df_kraken2 = pd.concat(kraken2_dfs)
    
    blastn_dfs = [pd.read_csv(file, sep="\\t", index_col=0) for file in files_blastn]
    df_blastn = pd.concat(blastn_dfs)
    summary_df = pd.join([df_kraken2, df_blastn])
    summary_df.to_csv("summary.tsv", sep="\\t")
```

Maybe you can try like this? Else we can talk again in person :)